packages/iov-keycontrol/src/keyring.spec.ts

```diff
@@ -126,6 +126,50 @@ describe("Keyring", () => {
         new Keyring(serialized);
       }).not.toThrow();
     });
+
+    it("throws when the same identity is added twice", async () => {
+      const wallet1 = Ed25519HdWallet.fromMnemonic(
+        "melt wisdom mesh wash item catalog talk enjoy gaze hat brush wash",
+      );
+      const wallet2 = Ed25519HdWallet.fromMnemonic(
+        "melt wisdom mesh wash item catalog talk enjoy gaze hat brush wash",
+      );
+      await wallet1.createIdentity(defaultChain, HdPaths.iov(0));
+      await wallet2.createIdentity(defaultChain, HdPaths.iov(0));
+
+      const keyring = new Keyring();
+      keyring.add(wallet1);
+      expect(() => keyring.add(wallet2)).toThrowError(/identity collision/i);
+    });
+  });
+
+  describe("createIdentity", () => {
+    it("works", async () => {
+      const keyring = new Keyring();
+      const wallet = keyring.add(
+        Ed25519HdWallet.fromMnemonic("melt wisdom mesh wash item catalog talk enjoy gaze hat brush wash"),
+      );
+
+      const newIdentity = await keyring.createIdentity(wallet.id, defaultChain, HdPaths.iov(0));
+
+      expect(keyring.getWallets()[0].getIdentities()).toEqual([newIdentity]);
+    });
+
+    xit("throws when the same identity is added twice", async () => {
```

Comment:
Is this supposed to be `xit`?

Reply:
The test was still failing when I commited it. It is active now 🚀